### PR TITLE
Adding parallel-from-list example

### PR DESCRIPTION
**Pull request by abayer**
cc @reviewbybees

### CODE REVIEW

File: parallel-from-list/parallelFromList.groovy [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+// still transform a list into a set of actual build steps to be executed in
+// parallel.
+
+// Our initial list of strings we want to echo in parallel
+def stringsToEcho = ["a", "b", "c", "d"]
+
+// The map we'll store the parallel steps in before executing them.
+def stepsForParallel = [:]
+
+// The standard 'for (String s: stringsToEcho)' syntax also doesn't work, so we
```
Comment (amuniz) [inline parallel-from-list/parallelFromList.groovy]: The nice syntax works if you wrap the code in a function annotated as `@NonCps` (you should not use regular steps inside the non-cps function, which is actually feasible in this example).

**Comment by amuniz** [comment]
:bee:

**Comment by oleg-nenashev** [comment]
:bee:

File: parallel-from-list/parallelFromList.groovy (hunk 1 above)
Comment (abayer) [inline parallel-from-list/parallelFromList.groovy]: Yeah, I'm working on figuring out a NonCps example I actually like. =)

File: parallel-from-list/parallelFromList.groovy [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+for (int i = 0; i < stringsToEcho.size(); i++) {
+    // Get the actual string here.
+    def s = stringsToEcho.get(i)
+
+    // Transform that into a step and add the step to the map as the value, with
+    // a name for the parallel step as the key. Here, we'll just use something
+    // like "echoing (string)"
+    def stepName = "echoing ${s}"
+    
+    stepsForParallel[stepName] = transformIntoStep(s)
```
Comment (jglick) [inline parallel-from-list/parallelFromList.groovy]: More simply:

```groovy
stepsForParallel[stepName] = {
  node {
    echo s
  }
}
```

**Pull request merged**